functions/src/https/api/map/index.ts

```diff
@@ -0,0 +1,52 @@
+import * as functions from 'firebase-functions';
+import { FieldPath } from '@google-cloud/firestore';
+
+import { db } from '../../../app';
+
+import { IMarker, MARKER_COLLECTION_ID } from '../../../models/markers';
+
+/* eslint-disable max-len */
+const LICENSES = {
+  reach4help:
+    'This data by Reach4Help (https://reach4help.org/) is licensed under a Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License (http://creativecommons.org/licenses/by-nc-sa/4.0/)',
+  // TODO: uncomment when license is confirmed with mutualaid.wiki
+  // 'mutualaid.wiki': `This data by Mutual Aid Wiki (https://mutualaid.wiki/) is licensed under a Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License (http://creativecommons.org/licenses/by-nc-sa/4.0/)`
+};
+/* eslint-enable max-len */
+
+type LicenseKey = keyof typeof LICENSES;
+
+interface Marker extends IMarker {
+  id: string;
+  license: LicenseKey;
+}
+
+interface Result {
+  licenses: typeof LICENSES;
+  data: Marker[];
+}
+
+// eslint-disable-next-line @typescript-eslint/no-misused-promises
+export const data = functions.https.onRequest(async (_req, res) => {
+  const result: Result = {
+    licenses: LICENSES,
+    data: [],
+  };
+  const markers = await db
+    .collection(MARKER_COLLECTION_ID)
+    .where(new FieldPath('visible'), '==', true)
+    .get();
+  markers.forEach(doc => {
+    const docData = doc.data() as IMarker;
+    // TODO: add other sources (with appropriate licenses)
+    if (!docData.source) {
+      result.data.push({
+        ...docData,
+        id: doc.id,
+        license: 'reach4help',
+      });
+    }
+  });
+  res.setHeader('Content-Type', 'application/json');
```

Comment:
```suggestion
  res.setHeader('Content-Type', 'application/json');
  res.setHeader('Cache-Control', 'public, max-age=300, s-maxage=600');
```

`max-age` is for browsers `s-maxage` is for the CDN.

Reply:
I suggest longer than `10` minutes, maybe a few hours at least.

Reply:
I would also suggest paginated replies, but that won't affect us so much as end user experience: https://firebase.google.com/docs/firestore/query-data/query-cursors

I'd recommend using the last returned document as the pagination token and simply `startAt(lastDocumentFromPreviousPage)` since document IDs are indexed and it's quite easy to combine that with filters.

Reply:
10 minutes is probably fine for now tbh, we want the data to be fresh :)

Pagination could be good, but given that this data is primarily going to be consumed by other projects in a data-sciency manner (rather than being used directly in frontend projects (at least I hope)), i think this is fine for now :)

Reply:
10 minutes is probably fine for now tbh, we want the data to be fresh :)

Pagination could be good, but given that this data is primarily going to be consumed by other projects in a data-sciency manner (rather than being used directly in frontend projects (at least I hope)), i think this is fine for now :)